### PR TITLE
Upgrade to CRA 2.1.3

### PR DESCRIPTION
**Pull request by DonghyukJacobJang**
- [x]  Upgrade to CRA 2.1.3
- [x]  Update component template
- [x]  Clean up `css` codes and integrate with `scss`
- [x]  Clean up SVG and use it as ReactComponent
- [x]  Replace `react-loadable` with `React.lazy` and `React.Suspense`
- [x]  Add webpack bundle analyzer
- [x] Add image compress script
- [x] eslint
- [ ] hot loader
## NOTE
- Removed `SvgComponent` folder under `src/components`. All SVG should be stored in `assets/svg`, and use it as `ReactComponent`
- Stateless component will use `React.forwardRef`.
- Added `npm run bundle-analyzer`
- Added `npm run image-compress` 

## PS
Please comment if I miss anything.

### CODE REVIEW

**Comment by DonghyukJacobJang** [comment]
Found that react-app-rewired is not compatible with CRA 2.x. Will need to find another way.

File: templates/src/components/Pages/Pages.js [hunk 1]
```diff
@@ -22,8 +22,24 @@ const Pages = ({ location, ...props }) => {
         <Transition appear key={location.pathname} timeout={getTransitionDuration(location.pathname)}>
           {state => (
             <Switch location={location}>
```
Comment (alemesa) [inline templates/src/components/Pages/Pages.js]: I think you can wrap Suspense completely over the the Switch @DonghyukJacobJang 
```javascript
<Suspense fallback={noop}>     
     <Switch location={location}> 
        <Route exact path={routeKeys.Landing} component={<Landing transitionState={state} />} />
        <Route exact path={routeKeys.About} component={<About transitionState={state} /> }/>
        <Route component={NotFound} />
    </Switch>
</Suspense>```

Comment (DonghyukJacobJang) [reply templates/src/components/Pages/Pages.js]: I can try that. Did not think of this idea. Thanks!

**Comment by DonghyukJacobJang** [comment]
For compressing images, we might need to do it as post processing.  Thinking this https://www.npmjs.com/package/imagemin

File: templates/src/pages/Landing/Landing.js [hunk 2]
```diff
@@ -3,21 +3,17 @@ import PropTypes from 'prop-types';
 import { connect } from 'react-redux';
 import classnames from 'classnames';
 
-import './Landing.css';
+import './Landing.scss';
 
 import BaseLink from '../../components/BaseLink/BaseLink';
 
+import { default as Transition } from '../PagesTransitionWrapper';
```
Comment (craighillwood) [inline templates/src/pages/Landing/Landing.js]: Maybe just use the custom name without the `{default as }` for "consistency" reasons.
```suggestion
import Transition from '../PagesTransitionWrapper';
```

File: templates/src/components/App/App.js [hunk 3]
```diff
@@ -20,6 +19,8 @@ import layout from '../../util/layout';
 import usePassiveEvent from '../../util/use-passive-event';
 import checkProps from '../../util/check-props';
 
+const RotateScreen = detect.isMobile ? lazy(() => import('../../components/Rotate/Rotate')) : null;
```
Comment (craighillwood) [inline templates/src/components/App/App.js]: I added the ternary `if` so `<RotateScreen>` could just be added to the `render` method (w/o checking there if `detec.isMobile`). But u are right that we need to wrap it in `<Suspense>` and for that u need to check again if `detect.isMobile`, so I think we can safely just change this to:
```suggestion
const RotateScreen = detect.isMobile && lazy(() => import('../../components/Rotate/Rotate'));
```
What do u think?

File: templates/src/components/App/App.js [hunk 4]
```diff
@@ -50,20 +51,24 @@ class App extends React.PureComponent {
 
   render() {
     return (
-      <Fragment>
+      <>
```
Comment (craighillwood) [inline templates/src/components/App/App.js]: Not so sure about using fragments as `<>`, do u know how the state of tooling is with this?
> Note that many tools don’t support it yet so you might want to explicitly write <React.Fragment> until the tooling catches up.

**Ref:** https://reactjs.org/docs/fragments.html#short-syntax

File: templates/src/components/Rotate/Rotate.js [hunk 5]
```diff
@@ -63,15 +63,17 @@ export default class RotateScreen extends PureComponent {
   render() {
     const visible = this.state.orientation === 'landscape';
     const style = {
-      visibility: visible ? 'visible' : 'hidden'
+      visibility: visible ? 'visible' : 'hidden' //
```
Comment (ragu-S) [inline templates/src/components/Rotate/Rotate.js]: Minor cleanup

File: templates/src/components/VideoPlayer/VideoTimeline/VideoTimeline.js [hunk 6]
```diff
@@ -38,7 +38,10 @@ export default class VideoTimeline extends React.PureComponent {
   };
 
   render() {
-    const progressStyle = { width: this.state.currentTime / this.props.duration * 100 + '%' };
+    const progressStyle = {
+      width: (this.state.currentTime / this.props.duration) * 100 + '%' //
```
Comment (ragu-S) [inline templates/src/components/VideoPlayer/VideoTimeline/VideoTimeline.js]: minor cleanup of `//`

File: templates/src/util/layout.js [hunk 7]
```diff
@@ -3,6 +3,13 @@ import { medium as MEDIUM, large as LARGE } from '../data/layout.json';
 export const MEDIUM_MEDIA_QUERY = `(min-width: ${MEDIUM}px)`;
 export const LARGE_MEDIA_QUERY = `(min-width: ${LARGE}px)`;
 
+// this code below is particularly for fixing jest test script
+window.matchMedia =
```
Comment (ragu-S) [inline templates/src/util/layout.js]: @DonghyukJacobJang shouldn't this be in the  `jest-setup.js`

**Pull request merged**